local_vars/local_vars.cc

```diff
@@ -70,7 +71,22 @@ class LocalNameInserter {
             }
             unique_ptr<ast::Reference> refExpImpl(refExp);
             arg.release();
-            namedArgs.emplace_back(nameArg(move(refExpImpl)));
+            auto named = nameArg(move(refExpImpl));
+            for (auto &prev : namedArgs) {
+                if (prev.name == named.name && !absl::StartsWith(prev.name.data(ctx)->shortName(ctx), "_")) {
+                    if (auto e = ctx.state.beginError(named.loc, core::errors::Namer::RepeatedArgument)) {
+                        auto frame = scopeStack.back();
+                        e.setHeader("Duplicated argument name `{}`", named.name.show(ctx));
+                        if (frame.insideMethod) {
```

Comment:
These extra clauses point to the method head (if the repeated arguments are in a method) or to the entire block (if in a block) because otherwise, for method heads that are spread across multiple lines, you might not get that context. For example, 
```ruby
def foo(
    this,
    this
); end
```
gives you the error message
```
fuzz/1133.rb:6: Duplicated argument name this https://srb.help/4017
     6 |      this:
              ^^^^^
    fuzz/1133.rb:4: In argument list of method
     4 |def foo(
     5 |      this,
     6 |      this:
     7 |    )
```

Reply:
Can you think of a case when we are neither in a block nor in a method and we get this error? Might be worth ENFORCE'ing in the else case if not.